flutter_example/lib/main.dart

```diff
@@ -4,29 +4,60 @@
 // `lib/main.declarations.dart` file:
 //
 //   - Instance of '_AutoDisposeMacro'
+//   - Instance of 'AutoListenable'
 //
 // To make changes you should edit the `lib/main.gen.dart` file;
 
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
-import 'macros/auto_dispose.dart';
+import 'macros/auto_listenable.dart';
 import 'macros/functional_widget.dart';
 import 'macros/render_accessors.dart';
 
 class MyHomePage extends StatefulWidget {
+  final ValueNotifier<int> counter;
   final String title;
   @override
   State<MyHomePage> createState() => _MyHomePageState();
-  const MyHomePage({Key? key, required this.title}) : super(key: key);
+  const MyHomePage(this.counter, {Key? key, required this.title})
+      : super(key: key);
 }
 
+@autoListenable
 class _MyHomePageState extends State<MyHomePage> {
-  final disposable = SimpleDisposable();
-  int _counter = 0;
   Color _color = Colors.blue;
+  @override
+  @autoListenable
+  void initState() {
+    super.initState();
+    widget.counter.addListener(_handleCounter);
+  }
+
+  @override
+  @autoListenable
+  void didUpdateWidget(
+    MyHomePage oldWidget,
+  ) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.counter != oldWidget.counter) {
+      oldWidget.counter.removeListener(_handleCounter);
+      widget.counter.addListener(_handleCounter);
+    }
+  }
+
+  @override
+  @autoListenable
+  void dispose() {
+    super.dispose();
```

Comment:
Place this at the _end_ of the method. [1]

[1] https://github.com/flutter/flutter/blob/497eb13d03f04a21125c6d8a3ee4f465a97e77e4/packages/flutter/lib/src/widgets/framework.dart#L1188

Reply:
Although, in practice the order for this particular case probably will not matter.